hawk/app/models/cib.rb

```diff
@@ -80,6 +81,119 @@ def sim?
     id != 'live'
   end
 
+  def status(minimal = false)
+    {
+      meta: meta.to_h,
+      errors: errors,
+      booth: booth
+    }.tap do |result|
+      if minimal
+        result[:resources] = {}
+
+        result[:resource_states] = {
+          pending: 0,
+          started: 0,
+          failed: 0,
+          master: 0,
+          slave: 0,
+          stopped: 0
+        }
+
+        result[:nodes] = {}
+
+        result[:node_states] = {
+          pending: 0,
+          online: 0,
+          standby: 0,
+          offline: 0,
+          unclean: 0
+        }
+
+        result[:tickets] = []
+
+        result[:ticket_states] = {
+          granted: 0,
+          revoked: 0
+        }
+
+        current_resources.each do |key, values|
+          result[:resources][key] ||= {}
+
+          values[:instances].each do |name, attrs|
+            found = false
+            [:master, :slave, :started, :failed, :pending].each do |rstate|
+              if attrs[rstate]
+                attrs[rstate].each { |n| result[:resources][key][n[:node]] = rstate }
+                result[:resource_states][rstate] += 1
+                found = true
+              end
+            end
+            result[:resource_states][:stopped] += 1 unless found
+          end
+        end
+
+        current_nodes.each do |node|
+          result[:nodes][node[:uname]] = node[:state]
+
+          result[:node_states][node[:state]] += 1
+        end
+
+        current_tickets.each do |key, values|
+          case
+          when values[:granted]
+            result[:tickets].push({ key => :granted })
+            result[:ticket_states][:granted] += 1
+          else
+            result[:tickets].push({ key => :revoked })
+            result[:ticket_states][:revoked] += 1
+          end
+        end
+
+        # TODO(must): Integrate tag listing
+        result[:tags] = []
+
+        # TODO(must): Integrate constraint listing
+        result[:constraints] = []
+      else
+        result[:crm_config] = crm_config
+        result[:rsc_defaults] = rsc_defaults
+        result[:op_defaults] = op_defaults
+
+        result[:resources] = resources
+        result[:nodes] = nodes
+        result[:tickets] = tickets
+        result[:tags] = tags
+        result[:constraints] = constraints
+      end
+    end
+  end
+
+  def current_resources
+    resources_by_id.select do |key, values|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.